File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/CommissionBasedPartTime .swift

public class CommissionBasedPartTime: PartTime {
    
    private var commision: Double
    
    override init() {
        commision = 0.0
        super.init()
    }
    
    init(ppName: String, ppAge: Int, pHourlyRate: Int, pNumberHoursWorked: Int, cbCommision: Double) {
        self.commision = cbCommision
        super.init(ppName: ppName, ppAge: ppAge, pHourlyRate: pHourlyRate, pNumberHoursWorked: pNumberHoursWorked)
        
    }
    
    init(ppName: String, ppAge: Int, pHourlyRate: Int, pNumberHoursWorked: Int, ppV: Vehicle, cbCommision: Double) {
        self.commision = cbCommision
        super.init(ppName: ppName, ppAge: ppAge, pHourlyRate: pHourlyRate, pNumberHoursWorked: pNumberHoursWorked, ppV: ppV)
    }
    
    private func calcSalary() -> Double {
        return Double(hourlyRate * numberHoursWorked)
    }
    
    private func calcCommision() -> Double {
        return calcSalary() * (commision / 100)
    }
    
    override func calcEarnings() -> Double {
        return calcSalary() + calcCommision()
    }
    
    override func getTypeOfEmployee() -> String {
        return super.getTypeOfEmployee() + " / Commissioned"
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -Commission: \(commision)%\n  -Earnings: \(calcEarnings()) (\(calcSalary()) + \(commision)% of \(calcSalary()))"
    }
    
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/employee/CommissionBasedPartTime.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.employee;

import c0711561.f2017.mad3464.finalproject.Util;
import c0711561.f2017.mad3464.finalproject.vehicle.Vehicle;
import java.math.BigDecimal;

/**
 *
 * @author macstudent
 */
public class CommissionBasedPartTime extends PartTime {
    
    private final double commission;

    public CommissionBasedPartTime(String name, int age, double rate, int hoursWorked, double commission) {
        super(name, age, rate, hoursWorked);
        this.commission = commission;
    }
    
    public CommissionBasedPartTime(String name, int age, double rate, int hoursWorked, double commission, Vehicle vehicle) {
        super(name, age, rate, hoursWorked, vehicle);
        this.commission = commission;
    }
    
    private double calcSalary() {
        return getRate() * getHoursWorked();
    }
    
    private double calcCommission() {
        return calcSalary() * (commission / 100);
    }

    @Override
    public String getTypeOfEmployee() {
        return super.getTypeOfEmployee() + " / Commissioned";
    }

    @Override
    public BigDecimal calcEarnings() {
        return new BigDecimal(commission).add(super.calcEarnings());
    }

    @Override
    public String printMyData() {
        return super.printMyData()
                + "\n  -Commission: " + commission
                + "\n  -Earnings: " + Util.toCurrencyFormatFrom(calcEarnings())
                + " " + calcSalary() + " + " + commission + "%" + " of " + calcSalary();
    }
}
